OathLab_Twitter/app/src/main/java/net/serkanbal/oathlab_twitter/MainActivity.java

```diff
@@ -0,0 +1,178 @@
+
+        mSearch.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                mScreenName = mScreenNameEntry.getText().toString();
+                getBearerToken(encodeKeys(CONSUMER_KEY, CONSUMER_SECRET));
+            }
+        });
+
+    }
+
+
+    // Encodes the consumer key and secret to create the basic authorization key
+    private String encodeKeys(String consumerKey, String consumerSecret) {
+        try {
+            String encodedConsumerKey = URLEncoder.encode(consumerKey, "UTF-8");
+            String encodedConsumerSecret = URLEncoder.encode(consumerSecret, "UTF-8");
+
+            String fullKey = encodedConsumerKey + ":" + encodedConsumerSecret;
+            byte[] encodedBytes = Base64.encodeBase64(fullKey.getBytes());
+            return new String(encodedBytes);
+        }
+        catch (UnsupportedEncodingException e) {
+            return null;
+        }
+    }
+
+    public void getBearerToken(String encoded) {
+        OkHttpClient client = new OkHttpClient();
+
+        RequestBody requestBody = new FormBody.Builder()
+                .add("grant_type", "client_credentials")
+                .build();
+
+        final Request request = new Request.Builder()
+                .addHeader("Authorization", "Basic " + encoded)
+                .addHeader("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
+                .post(requestBody)
+                .url("https://api.twitter.com/oauth2/token")
+                .build();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if (!response.isSuccessful()) {
+                    throw new IOException("on Response " + response);
+                } else {
+                    String data = response.body().string();
+
+                    Gson gson = new Gson();
+                    BearerToken bearer = gson.fromJson(data, BearerToken.class);
+                    String bearerToken = bearer.getBearerToken();
+
+//                        JSONObject obj = new JSONObject(response.body().string());
+//                        String bearerToken = obj.getString("access_token");
+                    Log.d("Access_Token", "Your Access Token is: " + bearerToken);
+
+                    getTweets(bearerToken, mScreenName);
+
+                }
+            }
+        });
+    }
+
+    private void getTweets(String bearerToken, String screenName) {
+        OkHttpClient client = new OkHttpClient();
+
+        Request request = new Request.Builder()
+                .addHeader("Authorization", "Bearer " + bearerToken)
+                .url("https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name=" +
```

Comment:
Might be helpful to put this url in a constant at the top of the class so it's easier to find